### ISSUE TITLE
Missing `isInstanceOf` information in concrete functions of ContractInterface

### ISSUE THREAD

**Issue by jad-hamza** [epfl-lara/smart#7]
This currently fails with a cast correctness issue in `getX`.

```scala
import stainless.smartcontracts._

trait A extends ContractInterface {
  var x: BigInt
  def getX() = x
}
```

**Comment by Frostweeds** [comment]
There was a bug with the injection of the precondition on the callee in ContractMethodLifting. Solved in #5

**Comment by jad-hamza** [comment]
Thanks!
